bin/translate-data.ts

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env ts-node-script
+
+/** Translate all of the files in `./data/` into properly tagged files in `./dist/data/` */
+
+import path from "path";
+import { promises as fs } from "fs";
+
+import toml from "@iarna/toml";
+
+async function main() {
+  const allData = await DataGroup.gatherFromDir("./data");
+  await allData.writeToDir("./dist/data");
+}
+
+class DataGroup {
+  typeAnnotation: string | null;
+  entries: Array<DataEntry>;
+  children: Record<string, DataGroup>;
+
+  constructor() {
+    this.typeAnnotation = null;
+    this.entries = [];
+    this.children = {};
+  }
+
+  static async gatherFromDir(sourcePath: string): Promise<DataGroup> {
```

Comment:
I'd probably use `glob` for this or something or else write a test for it, but i guess it's unlikely to change much

Reply:
There is already a `walk` function that can recursively go through a directory (in a `utils.ts` file). I didn't use it here because tying the recursive nature of the file tree directly to the data structure was very helpful, and I couldn't do that with something like an iterator over files/directories. I don't think `glob` would be any different, based on a quick scan of their docs.